proposals/NNNN-ios-check-type.md

```diff
@@ -0,0 +1,128 @@
+# Explicit returned type from NSDictrionary(Store) category
+
+* Proposal: [SDL-NNNN](NNNN-ios-check-type.md)
+* Author: [Misha Vyrko](https://github.com/mvyrko)
+* Status: **Awaiting review**
+* Impacted Platforms: [iOS]
+
+## Introduction
+
+This proposal is to specify returned types from NSDictionary (Store).
+This ensures that the return value is assigned to a variable of the correct type. 
+
+## Motivation
+
+Method (`-(nullable id)sdl_objectForName:(SDLName)name`) in NSDictionary (Store) returns `id` in the future it can be casted to any type.
+As result wrong casting leads to a crash in runtime.
+For example:
+````
+- (SDLLanguage)languageDesired {
+    id object = [parameters sdl_objectForName:SDLNameLanguageDesired];
+    return (SDLLanguage)object;
+} 
+
+- (void)someFunction:(SDLLanguage)language {
+    if ([self.languageDesired isEqualToEnum:language]) { // `-[__NSCFNumber isEqualToEnum:]: unrecognized selector sent to instance`
+        
+    }
+}
+````
+
+## Proposed solution
+
+Remove method :
+`-(nullable id)sdl_objectForName:(SDLName)name;`. 
+
+Instead of removed method use existing method:
```

Comment:
Please update this last one: 

Instead of the removed method, use this existing method: